src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs

```diff
@@ -0,0 +1,708 @@
+        public static void FloatingPointConstants_IncompatibleNumber()
+        {
+            AssertFloatingPointIncompatible_Fails<byte>();
+            AssertFloatingPointIncompatible_Fails<sbyte>();
+            AssertFloatingPointIncompatible_Fails<short>();
+            AssertFloatingPointIncompatible_Fails<int>();
+            AssertFloatingPointIncompatible_Fails<long>();
+            AssertFloatingPointIncompatible_Fails<ushort>();
+            AssertFloatingPointIncompatible_Fails<uint>();
+            AssertFloatingPointIncompatible_Fails<ulong>();
+            AssertFloatingPointIncompatible_Fails<float>();
+            AssertFloatingPointIncompatible_Fails<decimal>();
+            AssertFloatingPointIncompatible_Fails<byte?>();
+            AssertFloatingPointIncompatible_Fails<sbyte?>();
+            AssertFloatingPointIncompatible_Fails<short?>();
+            AssertFloatingPointIncompatible_Fails<int?>();
+            AssertFloatingPointIncompatible_Fails<long?>();
+            AssertFloatingPointIncompatible_Fails<ushort?>();
+            AssertFloatingPointIncompatible_Fails<uint?>();
+            AssertFloatingPointIncompatible_Fails<ulong?>();
+            AssertFloatingPointIncompatible_Fails<float?>();
+            AssertFloatingPointIncompatible_Fails<decimal?>();
+        }
+
+        private static void AssertFloatingPointIncompatible_Fails<T>()
+        {
+            string[] testCases = new[]
+            {
+                "NaN",
+                "Infinity",
+                "-Infinity",
+            };
+
+            foreach (string test in testCases)
+            {
+                Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<T>(test, s_optionReadFromStrAllowFloatConstants));
+            }
+        }
+
+        [Fact]
+        public static void UnsupportedFormats()
+        {
+            AssertUnsupportedFormatThrows<byte>();
+            AssertUnsupportedFormatThrows<sbyte>();
+            AssertUnsupportedFormatThrows<short>();
+            AssertUnsupportedFormatThrows<int>();
+            AssertUnsupportedFormatThrows<long>();
+            AssertUnsupportedFormatThrows<ushort>();
+            AssertUnsupportedFormatThrows<uint>();
+            AssertUnsupportedFormatThrows<ulong>();
+            AssertUnsupportedFormatThrows<float>();
+            AssertUnsupportedFormatThrows<decimal>();
+            AssertUnsupportedFormatThrows<byte?>();
+            AssertUnsupportedFormatThrows<sbyte?>();
+            AssertUnsupportedFormatThrows<short?>();
+            AssertUnsupportedFormatThrows<int?>();
+            AssertUnsupportedFormatThrows<long?>();
+            AssertUnsupportedFormatThrows<ushort?>();
+            AssertUnsupportedFormatThrows<uint?>();
+            AssertUnsupportedFormatThrows<ulong?>();
+            AssertUnsupportedFormatThrows<float?>();
+            AssertUnsupportedFormatThrows<decimal?>();
+        }
+
+        private static void AssertUnsupportedFormatThrows<T>()
```

Comment:
Are the formats tested in this method supported on Json.NET?

Reply:
No.

Reply:
Not by default - also can't find a custom option.